src/qt/paymentserver.cpp

```diff
@@ -414,7 +422,71 @@ void PaymentServer::handleURIOrFile(const QString& s)
 #else
         QUrlQuery uri((QUrl(s)));
 #endif
-        if (uri.hasQueryItem("r")) // payment request URI
+        if (uri.hasQueryItem("m") && uri.hasQueryItem("a")) // payment request URI
+        {
+            QByteArray temp;
+            temp.append(uri.queryItemValue("m"));
+            temp.append(uri.queryItemValue("a"));
+            QString decoded = QUrl::fromPercentEncoding(temp);
+            QUrl fetchUrl(decoded, QUrl::StrictMode);
+
+            if (!model)
```

Comment:
should we throw an error here?

Reply:
This is just a sanity check to prevent referencing a null pointer. Should never happen (as model is defined on startup). Gui code is full of checks like this and it never throws an error.